core/src/message/mod.rs

```diff
@@ -86,6 +87,62 @@ impl Payload {
     }
 }
 
+/// Buffer which size cannot be bigger then max allowed payload size.
+#[derive(Clone, Default, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct PanicBuffer(Payload);
+
+impl PanicBuffer {
+    /// Returns ref to the internal data.
+    pub fn inner(&self) -> &[u8] {
+        self.0.inner()
+    }
+
+    fn as_limited_str(&self) -> Option<LimitedStr> {
+        let s = core::str::from_utf8(self.0.inner()).ok()?;
```

Comment:
lossy?

Reply:
It means byte array always will be represented as mix of valid and invalid characters